src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -18,9 +18,24 @@
  *
  * @author Alex Bogomazov
  */
-abstract class FilterIterator extends \FilterIterator
+abstract class FilterIterator extends \FilterIterator implements \RecursiveIterator
```

Comment:
@nicolas-grekas but having an iterator configured to exclude ``/Tests$/`` and having it exclude ``.../Tests/foo.php`` would be confusing.
The filtering iterators are meant to implement the filters provided in the finder component, and implementing RecursiveIterator should be done only on filters which are actually providing their expected behavior in a recursive filtering IMO. Otherwise it would be too easy for someone to break the behavior in the future by moving filters around and getting one before the RecursiveIteratorIterator by mistake